axelor-stock/src/main/resources/views/StockMove.xml

```diff
@@ -210,7 +210,10 @@
                     <viewer>{{record.toAddressStr}}</viewer>
                 </field>
             </panel>
-
+            
+            <field name="$filterOnAvailableProducts" type="boolean" title="Filter on available products" onChange="action-stock-move-method-update-stock-move-line"/>
```

Comment:
Put this field also as transient instead of dummy and with default true same as on line.